File: 1-Two-Sum/solution.go
package main

func twoSum(nums []int, target int) []int {
	m := make(map[int]int, len(nums))

	for i, num := range nums {
		if value, exists := m[target-num]; exists {
			return []int{i, value}
		}
		m[num] = i
	}
	return nil
}

File: 1-Two-Sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const m = new Map();

  for (let i = 0; i < nums.length; i++) {
    const requiredNum = target - nums[i];

    if (m.has(requiredNum)) {
      return [m.get(requiredNum), i];
    }

    m.set(nums[i], i);
  }
};
